File: atomic-x/ios/Classes/AudioPlayer/AudioPlayerPlugin.swift
import Flutter
import UIKit

public class AudioPlayerPlugin: NSObject {
    private var methodChannel: FlutterMethodChannel?
    private var eventChannel: FlutterEventChannel?
    private var handler: AudioPlayerHandler?
    
    init(registrar: FlutterPluginRegistrar) {
        super.init()
        
        methodChannel = FlutterMethodChannel(
            name: "atomic_x/audio_player",
            binaryMessenger: registrar.messenger()
        )
        
        eventChannel = FlutterEventChannel(
            name: "atomic_x/audio_player_events",
            binaryMessenger: registrar.messenger()
        )
        
        handler = AudioPlayerHandler(
            methodChannel: methodChannel!,
            eventChannel: eventChannel!
        )
        
        methodChannel?.setMethodCallHandler { [weak self] (call, result) in
            self?.handler?.handle(call, result: result)
        }
        
        eventChannel?.setStreamHandler(handler)
    }
    
    public func dispose() {
        handler?.dispose()
        handler = nil
        methodChannel?.setMethodCallHandler(nil)
        eventChannel?.setStreamHandler(nil)
        methodChannel = nil
        eventChannel = nil
    }
}

File: atomic-x/android/src/main/kotlin/io/trtc/tuikit/atomicx/audioplayer/AudioPlayerPlugin.kt
package io.trtc.tuikit.atomicx.audioplayer

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodChannel

class AudioPlayerPlugin : FlutterPlugin {
    private var methodChannel: MethodChannel? = null
    private var eventChannel: EventChannel? = null
    private var handler: AudioPlayerHandler? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        methodChannel = MethodChannel(binding.binaryMessenger, "atomic_x/audio_player")
        eventChannel = EventChannel(binding.binaryMessenger, "atomic_x/audio_player_events")

        handler = AudioPlayerHandler(methodChannel!!, eventChannel!!)

        methodChannel?.setMethodCallHandler(handler)
        eventChannel?.setStreamHandler(handler)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        handler?.dispose()
        handler = null
        methodChannel?.setMethodCallHandler(null)
        eventChannel?.setStreamHandler(null)
        methodChannel = null
        eventChannel = null
    }
}
